### PR TITLE
tests: using the nested-state tool in nested tests

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This is the first step, then new commands will be migrated as well.

### CODE REVIEW

File: spread.yaml [hunk 1]
```diff
@@ -863,22 +863,12 @@ suites:
...
-            nested_destroy_vm
-            nested_cleanup_env
-
+            "$TESTSTOOLS"/nested-state remove-vm
+            "$TESTSTOOLS"/nested-state restore
             tests.backup restore
         restore: |
-            #shellcheck source=tests/lib/nested.sh
-            . "$TESTSLIB/nested.sh"
-            nested_cleanup_env
```
Comment (stolowski) [inline spread.yaml:880]: This is not needed because restore-each does it right?

File: spread.yaml [hunk 2]
```diff
@@ -962,27 +942,16 @@ suites:
...
-            nested_create_core_vm
+            "$TESTSTOOLS"/nested-state prepare
+            "$TESTSTOOLS"/nested-state build-image core
         prepare-each: |
             tests.backup prepare
-
-            #shellcheck source=tests/lib/nested.sh
-            . "$TESTSLIB/nested.sh"
-            nested_start_core_vm
-            nested_wait_for_snap_command
```
Comment (stolowski) [inline spread.yaml:975]: Ok, this is fine since we now wait inside nested_start_core_vm (nested_start_core_vm_unit).

File: spread.yaml [hunk 3]
```diff
@@ -1017,29 +986,16 @@ suites:
...
-
+            "$TESTSTOOLS"/nested-state remove-vm
             tests.backup restore
         restore: |
-            #shellcheck source=tests/lib/nested.sh
-            . "$TESTSLIB/nested.sh"
-            nested_destroy_vm
-            nested_cleanup_env
-
+            "$TESTSTOOLS"/nested-state restore
```
Comment (stolowski) [inline spread.yaml:998]: restore doesn't seem to be calling nested_destroy_vm, is this correct?

**Review by stolowski — COMMENTED** [review]
Thanks for this, looks fine, but one question.

File: spread.yaml (hunk 1 above)
Comment (bboozzoo) [reply spread.yaml:880]: There's a call to `nested-state restore` in other suites, so it's unclear whether we need it or not. Do we?

File: spread.yaml (hunk 2 above)
Comment (sergiocazzolato) [reply spread.yaml:975]: right

File: spread.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply spread.yaml:880]: The idea is that in the manual suite we create/restore the whole environment for each test. So as for each test we do "$TESTSTOOLS"/nested-state restore, then it not needed to do it when restoring the suite.
For other suites, on each test restore we do "$TESTSTOOLS"/nested-state remove-vm, so then when we restore the suite we need to do "$TESTSTOOLS"/nested-state restore.

File: spread.yaml (hunk 3 above)
Comment (sergiocazzolato) [reply spread.yaml:998]: Right, the restore cleans up the directories, but previously the VM needs to be removed.

Comment (stolowski) [reply spread.yaml:998]: Ok, I see "$TESTSTOOLS"/nested-state remove-vm (meaning nested_destroy_vm) is called by restore-each above.

**Review by stolowski — APPROVED** [review]
LGTM, thanks

**Pull request merged**